graylog2-server/src/main/java/org/graylog2/contentpacks/model/entities/NativeEntityDescriptor.java

```diff
@@ -29,26 +30,44 @@
 @AutoValue
 @JsonDeserialize(builder = AutoValue_NativeEntityDescriptor.Builder.class)
 public abstract class NativeEntityDescriptor implements Identified, Typed {
-    public static NativeEntityDescriptor create(ModelId id, ModelType type) {
+    public static final String FIELD_ENTITY_ID = "entity_id";
+
+    @JsonProperty(FIELD_ENTITY_ID)
+    public abstract ModelId entityId();
+
+    public static NativeEntityDescriptor create(ModelId entityId, ModelId id, ModelType type) {
         return builder()
+                .entityId(entityId)
                 .id(id)
                 .type(type)
                 .build();
     }
 
     /**
-     * Shortcut for {@link #create(ModelId, ModelType)}
+     * Shortcut for {@link #create(String, String, ModelType)}
      */
-    public static NativeEntityDescriptor create(String id, ModelType type) {
-        return create(ModelId.of(id), type);
+    public static NativeEntityDescriptor create(String entityId, String nativeId, ModelType type) {
+        return create(ModelId.of(entityId), ModelId.of(nativeId), type);
```

Comment:
How about creating a `NativeEntityDescriptor.create(ModelId entityId, ...)` method to avoid the `ModelId.of(entityId)` call? (that also removes some `entity.id().toString()` calls.